components/SearchBar/index.js

```diff
@@ -0,0 +1,201 @@
+import { useState } from "react";
+import TransactionCard from "../TransactionCard/index";
+import {
+  StyledFilterButton,
+  StyledFilterContainer,
+  StyledFilterItem,
+  StyledSearchAndFilter,
+  StyledSearchContainer,
+  StyledSearchInput,
+  StyledSearchInputSelect,
+  StyledSearchUl,
+} from "./SearchBar.styled";
+
+function SearchBar({ transactions, search, onSearch, isSearchEntry }) {
+  const [searchNameValue, setSearchNameValue] = useState(search.name);
+
+  const filterTransactions = () => {
+    return transactions.filter((transaction) => {
+      const dateFrom = search.dateFrom ? new Date(search.dateFrom) : null;
+      const dateTo = search.dateTo ? new Date(search.dateTo) : null;
+      const transactionDate = new Date(transaction.date);
+      const amountFrom = search.amountFrom
+        ? parseFloat(search.amountFrom)
+        : null;
+      const amountTo = search.amountTo ? parseFloat(search.amountTo) : null;
+      const transactionAmount = parseFloat(transaction.amount);
+
+      const nameMatches =
+        search.name === "" ||
+        transaction.name.toLowerCase().includes(search.name.toLowerCase());
+      const categoryMatches =
+        search.category === "" ||
+        transaction.category.toLowerCase() === search.category.toLowerCase();
+      const directionMatches =
+        search.direction === "" ||
+        transaction.direction.toLowerCase() === search.direction.toLowerCase();
+      const dateMatches =
+        !dateFrom ||
+        !dateTo ||
+        (transactionDate >= dateFrom && transactionDate <= dateTo);
+      const amountMatches =
+        !amountFrom ||
+        !amountTo ||
+        (transactionAmount >= amountFrom && transactionAmount <= amountTo);
+
+      return (
+        nameMatches &&
+        categoryMatches &&
+        directionMatches &&
+        dateMatches &&
+        amountMatches
+      );
+    });
+  };
+  const [isFilter, setIsFilter] = useState(false);
+
+  function handleResetFilters() {
+    onSearch({
+      name: "",
+      category: "",
+      direction: "",
+      dateFrom: "",
+      dateTo: "",
+      amountFrom: "",
+      amountTo: "",
+    });
+    setSearchNameValue("");
+  }
+
+  const handleKeyDown = (event) => {
+    if (event.key === " " && event.target.selectionStart === 0) {
+      event.preventDefault();
+    }
+  };
+
+  const handleSearchChange = (event) => {
+    const trimmedValue = event.target.value.trimStart();
```

Comment:
why don't you trim the end as well? `value.trim()`

Reply:
You should be able to enter multiple words, that's why we didn't do that.